### DISCUSSION TITLE
DEM vs CTX

### DISCUSSION THREAD

**Discussion by DmitriiSeverinov** [aertslab/pycistarget#32] [Q&A] [answered]
Dear all,

I am not quite sure if I understand correctly the meaning of CTX and DEM outputs for topics after running pycisTarget.
DEM is calculated based on a set of foreground regions that belong to a certain topic, and background regions are randomly(?) chosen regions from all other topics.
CTX is calculated based on the recovery curve approach.

The question is what type of "biology" both of them reflect?
Because I have a motor neurons population from the spinal cord that splits into two different subtypes. CTX and DEM results diverge significantly, such that for CTX I have expected motor neurons TFs in one subtype and HOX gens in another, while for DEM I have Yamanaka factors in one population and liver and retina specific TFs in another one.

Biologically speaking, to annotate the topic or understand the difference between cell subtypes I would go for CTX as it gives results that are more likely to be true. However, should it be always done this way, i.e., just looking at the results and choose the methods that fit best, or there is another approach to such a question?

Best,
Dmitrii

**Comment by SeppeDeWinter** [comment, marked as answer]
Hi @DmitriiSeverinov 

DEM (**D**ifferential **E**nrichment of **M**otifs) is designed to test for differential motif enrichment between two sets of regions. Think about it as differential gene expression analysis.

cistarget is designed to test what motifs are enriched in a set of regions compared to the whole genome (all consensus peaks in the case of SCENIC+).

For that reason it is normal that both give different results, because they are very different algorithms. 

As to choose a method, I would do it based on the question. If you want to see what is different between let's say topic 1 and topic 2; use DEM. If you are not interested in the difference of two topics, use cistarget.

I hope this helps?

All the best,

Seppe

**Comment by DmitriiSeverinov** [comment]
Hi @SeppeDeWinter ,

Thank you very much for the clarification!
So and if I get this correctly, DEM by default makes a comparison one vs all, e.g., topic 1 vs topic 2,3,4,5. If I am interested in topic 1 vs topic 2 I need to do such a comparison separately, is it correct?
My motivation for this is that I want to find out the difference in enriched motifs between two motor neuron populations. This is why I am not really interested in comparing one population vs all other cell types, but rather having motor neurons 1 vs motor neurons 2.

**Reply by SeppeDeWinter** [reply]
Hi @DmitriiSeverinov 

That's correct, if you follow the SCENIC+ tutorial. 
For the SCENIC+ tutorial in fact the folder structure dictates what comparisons will be made. See example below

```bash

DARs
|______ Motor_neuron_1.bed
|______ Motor_neuron_2.bed
|______ Cell_type_1.bed
|______ Cell_type_2.bed
|______ Cell_type_3.bed
|______ Cell_type_4.bed
|______ Cell_type_5.bed

```

Will do motor neuron 1 vs motor neuron 2, cell type 2, 3, 4, 5 (and all other combinations)

While

```bash

DARs_1
|______ Motor_neuron_1.bed
|______ Motor_neuron_2.bed

DARs_2
|______ Cell_type_1.bed
|______ Cell_type_2.bed
|______ Cell_type_3.bed
|______ Cell_type_4.bed
|______ Cell_type_5.bed

```

Will do motor neuron 1 vs motor neuron 2 (and 2 vs 1) and cell type 1 vs 2, 3, 4, 5 (and all other combinations).

Seppe

**Reply by DmitriiSeverinov** [reply]
Thanks, now it got very clear!
